src/main/java/mil/dds/anet/resources/AdminResource.java

```diff
@@ -104,52 +105,48 @@ public String clearCache(@GraphQLRootContext Map<String, Object> context) {
-        // the other one is "recentCalls" which holds all entries without grouping them
+        // Group all entries with recentCalls key
         recentCalls.computeIfAbsent("recentCalls", l -> new LinkedHashSet<>()).add(k);
       });
     }
-    final Map<String, Object> allActivities = new HashMap<>();
-    // Sort userActivities by time
-    if (!userActivities.isEmpty()) {
-      allActivities.put("users", userActivities.values().stream().map(s -> {
-        final ArrayList<Map<String, String>> activities = new ArrayList<>(s);
-        final List<Map<String, String>> sortedList =
-            activities.stream().sorted((o1, o2) -> o2.get("time").compareTo(o1.get("time")))
-                .collect(Collectors.toList());
-        return new LinkedHashSet<>(sortedList);
-      }).collect(Collectors.toMap(k -> k.iterator().next().get("user"), Function.identity())));
-    } else {
-      allActivities.put("users", "");
-    }
-    // Sort recentCalls by time
-    if (!recentCalls.isEmpty()) {
-      allActivities.put("recentCalls", recentCalls.values().stream().map(s -> {
-        final ArrayList<Map<String, String>> activities = new ArrayList<>(s);
-        final List<Map<String, String>> sortedList =
-            activities.stream().sorted((o1, o2) -> o2.get("time").compareTo(o1.get("time")))
-                .collect(Collectors.toList());
-        return new LinkedHashSet<>(sortedList);
-      }).collect(Collectors.toMap(k -> k.iterator().next().get("activity"), Function.identity()))
-          .get("activity"));
-    } else {
-      allActivities.put("recentCalls", "");
-    }
+
+    if (recentCalls.size() == 0)
+      return new HashMap<>();
+
+    final HashMap<String, Object> allActivities = new HashMap<>();
+    final HashMap<String, LinkedHashSet<Map<String, String>>> recentActivities =
+        new HashMap<String, LinkedHashSet<Map<String, String>>>() {
+          {
+            // Sort recentCalls by time in descending order
+            put("recentCalls", recentCalls.values().stream().map(s -> {
+              final ArrayList<Map<String, String>> activities = new ArrayList<>(s);
+              return activities.stream()
+                  .sorted((o1, o2) -> o2.get("time").compareTo(o1.get("time")))
+                  .collect(Collectors.toCollection(LinkedHashSet::new));
+            }).collect(
+                Collectors.toMap(k -> k.iterator().next().get("activity"), Function.identity()))
+                .get("activity"));
+          }
+        };
+    allActivities.put("recentCalls", recentActivities.get("recentCalls"));
+
+    // Sort recentActivities grouping by user
+    final LinkedHashMap<String, List<Map<String, String>>> recentUsers = recentActivities
+        .get("recentCalls").stream().collect(Collectors.groupingBy(item -> item.get("user"),
+            LinkedHashMap::new, Collectors.mapping(Function.identity(), Collectors.toList())));
+
+    // Put sorted users to the list
```

Comment:
This comment reads exactly like the code below, to me, an unnecessary comment is more of a distraction rather than useful information/explanation.